static/js/src/advantage/subscribe/react/components/Summary.jsx

```diff
@@ -0,0 +1,139 @@
+import React from "react";
+import { Row, Col } from "@canonical/react-components";
+import { add, format } from "date-fns";
+import { formatter } from "../../renderers/form-renderer";
+import usePreview from "../APICalls/usePreview";
+import useProduct from "../APICalls/useProduct";
+
+const DATE_FORMAT = "dd MMMM yyyy";
+
+function Summary() {
+  const { product, quantity } = useProduct();
+  const { data: preview } = usePreview();
+
+  return (
+    <section
+      id="summary-section"
+      className="p-strip is-shallow u-no-padding--top"
+    >
+      <Row className="u-no-padding u-sv1">
+        <Col size="4">
+          <div className="u-text-light">Plan type:</div>
+        </Col>
+        <Col size="8">
+          <div>{product?.name}</div>
+        </Col>
+      </Row>
+      <Row className="u-no-padding u-sv1">
+        <Col size="4">
+          <div className="u-text-light">Machines:</div>
+        </Col>
+        <Col size="8">
+          <div>
+            {quantity} x {formatter.format(product?.price?.value / 100)}
+          </div>
+        </Col>
+      </Row>
+      <Row className="u-no-padding u-sv1">
+        <Col size="4">
+          <div className="u-text-light">Starts:</div>
+        </Col>
+        <Col size="8">
+          <div>{format(new Date(), DATE_FORMAT)}</div>
+        </Col>
+      </Row>
+      <Row className="u-no-padding u-sv1">
+        <Col size="4">
+          <div className="u-text-light">Ends:</div>
+        </Col>
+
+        {preview?.subscriptionEndOfCycle ? (
+          <Col size="8">
+            {format(new Date(preview?.subscriptionEndOfCycle), DATE_FORMAT)}
+            <br />
+            <small>The same date as your existing subscription.</small>
+          </Col>
+        ) : (
+          <Col size="8">
+            <div>
+              {format(
+                add(new Date(), {
+                  months: product?.period === "monthly" ? 1 : 12,
+                }),
+                DATE_FORMAT
+              )}
+            </div>
+          </Col>
+        )}
+      </Row>
+      {preview ? (
```

Comment:
This is a bit of an excessive use of a ternary :)

Consider abstracting this logic out into separate render functions.